File: MineInAbyss/src/main/java/com/derongan/minecraft/mineinabyss/ascension/AscensionCommandExecutor.java
package com.derongan.minecraft.mineinabyss.ascension;

import com.derongan.minecraft.mineinabyss.AbyssContext;
import com.derongan.minecraft.mineinabyss.Permissions;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;
import org.bukkit.permissions.Permission;

public class AscensionCommandExecutor implements CommandExecutor {
    private AbyssContext context;

    public AscensionCommandExecutor(AbyssContext context) {
        this.context = context;
    }

    @Override
    public boolean onCommand(CommandSender commandSender, Command command, String label, String[] args) {
        if (commandSender instanceof Player) {
            Player player = (Player) commandSender;

            if(!player.hasPermission(Permissions.TOGGLE_CURSE)){
                player.sendMessage(ChatColor.RED + "The abyss laughs at your naivety");
                return true;
            }

            if (label.equals("curseon") && player.hasPermission(Permissions.TOGGLE_CURSE)) {
                context.getPlayerDataMap().get(player.getUniqueId()).setAffectedByCurse(true);
                player.sendMessage("Curse enabled");
                return true;
            }

            if (label.equals("curseoff") && player.hasPermission(Permissions.TOGGLE_CURSE)){
                context.getPlayerDataMap().get(player.getUniqueId()).setAffectedByCurse(false);
                player.sendMessage("Curse disabled");
                return true;
            }
        }

        return false;
    }
}

File: src/main/java/com/derongan/minecraft/mineinabyss/ascension/AscensionCommandExecutor.kt
package com.derongan.minecraft.mineinabyss.ascension

import com.derongan.minecraft.mineinabyss.commands.CommandLabels
import com.derongan.minecraft.mineinabyss.mineInAbyss
import com.derongan.minecraft.mineinabyss.playerData
import com.mineinabyss.idofront.commands.IdofrontCommandExecutor
import com.mineinabyss.idofront.commands.onExecuteByPlayer
import com.mineinabyss.idofront.messaging.success

class AscensionCommandExecutor : IdofrontCommandExecutor() {
    override val commands = commands(mineInAbyss) {
        shared {
            noPermissionMessage = "The abyss laughs at your naivety"
        }

        command(CommandLabels.CURSEON) {
            onExecuteByPlayer {
                player.success("Curse enabled")
                player.playerData.isAffectedByCurse = true
            }
        }

        command(CommandLabels.CURSEOFF) {
            onExecuteByPlayer {
                player.success("Curse disabled")
                player.playerData.isAffectedByCurse = false
            }
        }
    }
}